agent/check_handler.go

```diff
@@ -401,11 +410,101 @@ func evaluateOutputMetricThresholds(event *corev2.Event) uint32 {
 		}
 		if !ruleMatched {
 			for _, rule := range thresholdRule.Thresholds {
-				if rule.NullStatus > 0 && status < rule.NullStatus {
-					status = rule.NullStatus
+				if rule.NullStatus > 0 {
+					addNullStatusThresholdAnnotation(event, thresholdRule, rule.NullStatus)
+					if status < rule.NullStatus {
+						status = rule.NullStatus
+						annotationValue = getNullStatusAnnotationValue(thresholdRule)
+					}
 				}
 			}
 		}
 	}
+
+	if annotationValue != "" {
+		event.AddAnnotation("sensu.io/notifications/"+corev2.CheckStatusToCaption(status), annotationValue)
+	}
+
 	return status
 }
+
+func addThresholdAnnotation(event *corev2.Event, metricThreshold *corev2.MetricThreshold, measure string, status uint32, value float64, threshold string) {
+	event.AddAnnotation(getAnnotationKey(metricThreshold, measure, status), getAnnotationValue(metricThreshold, measure, value, threshold))
+}
+
+func addNullStatusThresholdAnnotation(event *corev2.Event, metricThreshold *corev2.MetricThreshold, status uint32) {
+	event.AddAnnotation(getAnnotationKey(metricThreshold, measureNullStatus, status), getNullStatusAnnotationValue(metricThreshold))
+}
+
+func getAnnotationKey(metricThreshold *corev2.MetricThreshold, measure string, status uint32) string {
+	var key strings.Builder
+
+	key.WriteString("sensu.io/output_metric_thresholds/")
+	key.WriteString(metricThreshold.Name)
+	for _, tag := range metricThreshold.Tags {
+		key.WriteString(".")
+		key.WriteString(tag.Value)
```

Comment:
I can't think of any serious gotchas here so I'll defer to your judgement, but are we concerned at all about stability of this annotation key?

So far I've seen sensu do a good job of not reordering lists, but through operator intervention or unspecified behavior we could end up going from `sensu.io/output_metric_thresholds/my_threshold.foo.bar/null-status/critical` for tags `source=foo,dest=bar` to `sensu.io/output_metric_thresholds/my_threshold.bar.foo/null-status/critical` pretty easily.

Reply:
Very good point Christian, I did overlook that aspect. I'll make the change to sort the tags by key.

Reply:
@c-kruse - now that I was about to fix this I believe this is actually not an issue. The tags are stored in a slice, not a map. This makes sure the order is maintained. The user would have to explicitly change the order using sensuctl for your scenario to happen. But for any given evaluation of an event the order would remain the same across annotation.

@calebhailey do you have any input on this issue?

Reply:
@fguimond nope.

Reply:
@calebhailey Sorry my question wasn't clear in this case. Should we sort the annotation before adding them to annotations or let them in the same order they were created?

In other words if we have the following scenario
```
spec:
  ...: ...
  output_metric_thresholds:
  - name: example_metric
    tags:
      - name: zoo
        value: none
      - name: foo
        value: bar
      - name: too
        value: dive
    thresholds:
      - max: "2.0"
        min: "1.0"
        null_status: 1
        status: 1
```
Do we want to output in sequential order
```
sensu.io/output_metric_thresholds/example_metric.none.bar.dive/...
```
or sorted by key
```
sensu.io/output_metric_thresholds/example_metric.bar.dive.none/...
```

I would personally lean toward sequential order but wanted to get your opinion.

Thanks!